src/cls/rbd/cls_rbd_client.cc

```diff
@@ -149,6 +149,18 @@ namespace librbd {
                                          parent);
     }
 
+    void create_image(librados::ObjectWriteOperation *op, uint64_t size, uint8_t order,
```

Comment:
I'd leave all the sync versions that are still needed by librbd -- just have the sync version call the async version to setup the object operation.